test/src/main/java/com/coderanch/test/AbstractObjectTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.test;
+
+import org.junit.Rule;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theory;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeNotNull;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * Tests the common methods of all objects.
+ */
+@SuppressWarnings("squid:S00100")
+public abstract class AbstractObjectTest {
+
+    /**
+     * A rule that states the kind of exception that is expected to be thrown by a test.
+     */
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    /**
+     * Tests that an object is never equal to null.
+     */
+    @Theory
+    public final void equals_withNull_returnsFalse(
+        @FromDataPoints("objects") Object x
+    ) {
+        assumeNotNull(x);
+        assertThat(x, is(not(equalTo(null))));
+    }
+
+    /**
+     * Tests that an object is always equal to itself.
+     */
+    @Theory
+    public final void equals_isReflexive(
+        @FromDataPoints("objects") Object x
+    ) {
+        assertThat(x, is(equalTo(x)));
+    }
+
+    /**
+     * Tests that two objects agree with each other on whether they're equal.
+     */
+    @Theory
+    public final void equals_isSymmetric(
+        @FromDataPoints("objects") Object x,
+        @FromDataPoints("objects") Object y
+    ) {
+        assumeNotNull(x, y);
+        assertThat(x.equals(y), is(equalTo(y.equals(x))));
+    }
+
+    /**
+     * Tests that three objects agree with each other on whether they're equal.
+     */
+    @Theory
+    public final void equals_isTransitive(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Document the parameter(s): x, y, z ([squid:UndocumentedApi](https://sonarcloud.io/organizations/coderanch-corral/rules#rule_key=squid%3AUndocumentedApi))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.coderanch%3Ablackjack-parent&pullRequest=11&issues=AWaB526EksgJibV5mvFl&open=AWaB526EksgJibV5mvFl)

[ISSUE_KEY]: AWaB526EksgJibV5mvFl